src/compiler/emitter.ts

```diff
@@ -2783,18 +2791,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
                     emitDestructuring(node, node.parent.kind === SyntaxKind.ExpressionStatement);
                 }
                 else {
-                    const exportChanged =
+                    const externalExportChanged =
                         node.operatorToken.kind >= SyntaxKind.FirstAssignment &&
                         node.operatorToken.kind <= SyntaxKind.LastAssignment &&
                         isNameOfExportedSourceLevelDeclarationInSystemExternalModule(node.left);
 
-                    if (exportChanged) {
+                    if (externalExportChanged) {
                         // emit assignment 'x <op> y' as 'exports("x", x <op> y)'
                         write(`${exportFunctionForFile}("`);
                         emitNodeWithoutSourceMap(node.left);
                         write(`", `);
                     }
 
+                    const internalExportClauseMemberChanged =
+                        node.operatorToken.kind >= SyntaxKind.FirstAssignment &&
+                        node.operatorToken.kind <= SyntaxKind.LastAssignment &&
+                        isNameOfExportedSourceLevelDeclarationInClauseModule(node.left);
+
+                    if (internalExportClauseMemberChanged) {
+                        for (const specifier of exportSpecifiers[(<Identifier>node.left).text]) {
+                            emitStart(specifier.name);
+                            emitContainingModuleName(specifier);

```

Comment:
I don't quite get this part, why does ES3 `exports["default"]` has anything to do since we're changing `exports` members?